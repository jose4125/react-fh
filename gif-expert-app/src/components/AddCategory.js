import { useState } from "react";
import PropTypes from "prop-types";

export default function AddCategory({ onAddCategory }) {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim()) {
      onAddCategory(inputValue);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChange} />
      <button type="submit">Add category</button>
    </form>
  );
}

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
