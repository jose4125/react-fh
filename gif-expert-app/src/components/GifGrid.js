import PropTypes from "prop-types";
import GifGridItem from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export default function GifGrid({ category }) {
  const [images, loading] = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {images && images.map((img) => <GifGridItem key={img.id} {...img} />)}
      </div>
    </>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
