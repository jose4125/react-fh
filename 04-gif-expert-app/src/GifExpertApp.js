import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

export default function GifExpertApp({ defaultCategories = [] }) {
  const [categories, setCategory] = useState(defaultCategories);

  const handleAddCategory = (category) => {
    setCategory([category, ...categories]);
  };

  return (
    <>
      <h2>gif expert app</h2>
      <AddCategory onAddCategory={handleAddCategory} />
      <hr />
      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
}
