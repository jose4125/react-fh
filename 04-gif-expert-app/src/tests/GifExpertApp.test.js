import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";
import GifGrid from "../components/GifGrid";
import AddCategory from "../components/AddCategory";

describe("GifGrid tests", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<GifExpertApp />);
  });

  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a title", () => {
    expect(wrapper.find("h2").exists()).toBe(true);
    expect(wrapper.find("h2").text()).toBe("gif expert app");
  });

  test("should show category list", () => {
    const categories = ["goku", "caballeros del zodiaco"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper.find("GifGrid").exists()).toBe(true);
    expect(wrapper.find("GifGrid")).toHaveLength(categories.length);
  });
});
