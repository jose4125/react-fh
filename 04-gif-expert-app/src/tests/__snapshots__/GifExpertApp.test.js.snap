// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should render properly 1`] = `
<Fragment>
  <h2>
    gif expert app
  </h2>
  <AddCategory
    onAddCategory={[Function]}
  />
  <hr />
</Fragment>
`;
